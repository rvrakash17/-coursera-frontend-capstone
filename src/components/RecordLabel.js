import Band from "./Band";

/**
 * Component for showing details of the RecordLabel.
 *
 * @component
 */
export default function RecordLabel(props) {
  return (
    <section>
      <h4 data-testid="recordlabel-label-name">{props.label.name ? props.label.name : "(independent bands)"}</h4>
      <ul>
        {props.label.bands.map(function (band, index) {
          return <Band band={band} key={index} />;
        })}
      </ul>
    </section>
  );
}
