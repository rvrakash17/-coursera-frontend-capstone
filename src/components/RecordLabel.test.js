import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import RecordLabel from "./RecordLabel";

describe("RecordLabel Component", () => {
  // Test 1
  test("RecordLabel Rendering", () => {
    /* Fixtures */
    const name = "Winter Primates";
    const label = {
      name: name,
      bands: [
          {
              name: "Frank Jupiter",
              festivals: [
                  {
                      "name": "LOL-palooza"
                  }
              ]
          },
          {
              name: "Propeller",
              festivals: [
                  {}
              ]
          }
      ]
  };
    render(<RecordLabel label={label} key={0} />);
    const recordlabel_label_name = screen.getByTestId("recordlabel-label-name");
    expect(recordlabel_label_name).toBeInTheDocument();

    // snapshot
    const domTree = renderer.create(<RecordLabel label={label} key={0} />).toJSON();
    expect(domTree).toMatchSnapshot();
  });

});
