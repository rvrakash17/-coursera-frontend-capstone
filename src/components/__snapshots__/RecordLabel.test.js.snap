// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Band Component Band Rendering with empty festivals 1`] = `
<li>
  <span
    data-testid="band-band-name"
  >
    Winter Primates
  </span>
  <ul />
</li>
`;

exports[`Band Component Band Rendering with missing festivals 1`] = `
<li>
  <span
    data-testid="band-band-name"
  >
    Winter Primates
  </span>
  <ul />
</li>
`;

exports[`Band Component Band Rendering with name and festivals 1`] = `
<li>
  <span
    data-testid="band-band-name"
  >
    Winter Primates
  </span>
  <ul>
    <li
      data-testid="band-festival-name"
    >
      LOL-palooza
    </li>
    <li
      data-testid="band-festival-name"
    >
      MIDI
    </li>
  </ul>
</li>
`;

exports[`RecordLabel Component RecordLabel Rendering 1`] = `
<section>
  <h4
    data-testid="recordlabel-label-name"
  >
    Winter Primates
  </h4>
  <ul>
    <li>
      <span
        data-testid="band-band-name"
      >
        Frank Jupiter
      </span>
      <ul>
        <li
          data-testid="band-festival-name"
        >
          LOL-palooza
        </li>
      </ul>
    </li>
    <li>
      <span
        data-testid="band-band-name"
      >
        Propeller
      </span>
      <ul>
        <li
          data-testid="band-festival-name"
        />
      </ul>
    </li>
  </ul>
</section>
`;
